Sources/InternxtSwiftCore/Services/Network/NetworkFacade.swift

```diff
@@ -132,40 +132,56 @@ public struct NetworkFacade {
         let uploadedPartsActor = UploadedPartsActor()
         let uploadState = UploadState()
         
-        var tasks: [Data] = []
+        input.open()
+        defer { input.close() }
+        
+        let operationQueue = OperationQueue()
+        operationQueue.maxConcurrentOperationCount = 6
+
+        var partIndex = 0
         try await encrypt.encryptFileIntoChunks(
             chunkSizeInBytes: MULTIPART_CHUNK_SIZE,
             totalBytes: fileSize,
             inputStream: input,
             key: fileKey,
             iv: iv
         ) { encryptedChunk in
-            hasher.update(data: encryptedChunk)
-            tasks.append(encryptedChunk)
-        }
-        
-        let operationQueue = OperationQueue()
-        operationQueue.maxConcurrentOperationCount = 6
-        
-        for (index, encryptedChunk) in tasks.enumerated() {
-            if await uploadedPartsActor.isPartUploaded(partIndex: index) {
-                continue
+            guard await !uploadState.isAborted() else {
+                throw UploadError.UploadNotSuccessful
             }
-            
-            let uploadUrl = uploadUrls[index]
+
+            if await uploadedPartsActor.isPartUploaded(partIndex: partIndex) {
```

Comment:
If this function is being called for each encrypted chunk, and as far as I see, it is not parallelized, why is this required? We should not aim to skip any chunk but wait to enqueue the next as you do a bit below correctly.